lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
@xaque208 Sorry, I was sick the last few days so I didn't get a chance to reply yet.  Generally we don't want logic that is specific to a provider represented as a fact; we feel this clutters facter with a bunch of facts that have no other use than being queried in the provider, especially when the provider could simply have performed the same work as facter did.

Here's one way I think I would fix this:

Change `confine :pkgng_enabled => :true` to:

    confine :true do
        # Perform the same work that the pkgng_enabled custom fact is doing; i.e. run pkg and see if it's the right one.
     end

This confine should only get executed when the previous confine returns true (i.e. only on FreeBSD and DragonFly).  The `defaultfor :pkgng_enabled => :true` line is then redundant as we've already confined the provider to where we know it's enabled.  There may have to be some massaging of the provider's `specificity` to make it work properly, but I don't see why this particularly requires a fact.  For example, the `ps` fact is only used in the base service provider.  We're removing this fact in Facter 3 and Puppet 4 is changing the provider to "do the right thing".